### PR TITLE
Support generating top level widget options and use it for popup height and width in kickoff

### PR DESCRIPTION
**Pull request by networkException**
This pull request adds support for generating JavaScript to configure top level widget settings and uses this to allow configuring kickoff's `popupHeight` and `popupWidth`.

For

```nix
programs.plasma.panels = [
  {
    widgets = [
      {
        kickoff = {
          popupHeight = 510;
          popupWidth = 778;
        };
      }
    ];
  }
];
```

the resulting JavaScript is

```js
panelWidgets["org.kde.plasma.kickoff"] = panel.addWidget("org.kde.plasma.kickoff");
var w = panelWidgets["org.kde.plasma.kickoff"];
w.currentConfigGroup = [];
w.writeConfig("popupHeight", 510);
w.writeConfig("popupWidth", 778);
```

and the config looks like this

```ini
[Containments][1694][Applets][1695][Configuration]
popupHeight=510
popupWidth=778
# ...

[Containments][1694][Applets][1695][Configuration][General]
# ...
```

### CODE REVIEW

File: modules/widgets/kickoff.nix [hunk 1]
```diff
@@ -83,6 +83,16 @@ in
...
+        example = 500;
+      };
+      popupWidth = mkOption {
+        type = with types; ints.positive;
+        default = null;
+        example = 700;
+      };
       settings = mkOption {
         type = with types; nullOr (attrsOf (attrsOf (either (oneOf [ bool float int str ]) (listOf (oneOf [ bool float int str ])))));
         default = null;
```
Comment (HeitorAugustoLN) [inline modules/widgets/kickoff.nix]: Settings option should be updated so they can also use top level widget options (for every single widget).

```suggestion
      settings = mkOption {
        type = with types; nullOr (attrsOf (either
          (attrsOf (either (oneOf [ bool float int str ]) (listOf (oneOf [ bool float int str ]))))
          (either (oneOf [ bool float int str ]) (listOf (oneOf [ bool float int str ])))
        ));
        default = null;
```

File: modules/widgets/kickoff.nix [hunk 2]
```diff
@@ -83,6 +83,16 @@ in
...
+      popupHeight = mkOption {
+        type = with types; ints.positive;
+        default = null;
+        example = 500;
+      };
+      popupWidth = mkOption {
+        type = with types; ints.positive;
+        default = null;
+        example = 700;
+      };
```
Comment (HeitorAugustoLN) [inline modules/widgets/kickoff.nix]:
```suggestion
      popupHeight = mkOption {
        type = with types; nullOr ints.positive;
        default = null;
        example = 500;
      };
      popupWidth = mkOption {
        type = with types; nullOr ints.positive;
        default = null;
        example = 700;
      };
```

**Comment by HeitorAugustoLN** [comment]
Sorry for the excessive amount of commits, I had to use github to edit the files, and I just noticed the options already had nullOr, so I will fix it later. But it is working great now after some changes I've made. Thank you for your contribution!

**Review by HeitorAugustoLN — APPROVED** [review]

**Comment by networkException** [comment]
Neat, thanks

**Pull request merged**